internal/dag/builder_test.go

```diff
@@ -14958,6 +15090,9 @@ func listeners(ls ...*Listener) []*Listener {
 func prefixString(prefix string) MatchCondition {
 	return &PrefixMatchCondition{Prefix: prefix, PrefixMatchType: PrefixMatchString}
 }
+func exactString(exact string) MatchCondition {
```

Comment:
nit: the `prefixString` was named since it represented a prefix match that would basically match a path using "string prefix" vs. matching path segments, we could change this to just `exact` here

Reply:
Makes sense, rather `exact()` is already there making it completely redundant.